reana_server/scheduler.py

```diff
@@ -125,67 +135,106 @@ def get_consumers(self, Consumer, channel):
             )
         ]
 
-    def on_message(self, workflow_submission, message):
+    def _fail_workflow(self, workflow_id: str, logs: str = "") -> None:
+        self.workflow_status_publisher.publish_workflow_status(
+            workflow_id, status=RunStatus.failed.value, logs=logs,
+        )
+
+    def _retry_submission(self, workflow_id: str, workflow_submission: Dict) -> None:
+        retry_count = workflow_submission.get("retry_count", 0)
+        if retry_count >= REANA_SCHEDULER_REQUEUE_COUNT:
+            error_message = (
+                f"Workflow {workflow_submission['workflow_id_or_name']} failed to schedule after "
+                f"{retry_count} retries. Giving up."
+            )
+            logging.error(error_message)
+            self._fail_workflow(workflow_id, logs=error_message)
+        else:
+            workflow_submission["retry_count"] = retry_count + 1
+            workflow_submission["user_id"] = workflow_submission["user"]
+            del workflow_submission["user"]
+            current_workflow_submission_publisher.publish_workflow_submission(
```

Comment:
Minor: alternatively we could specify all the arguments like it was done [here](https://github.com/reanahub/reana-server/blob/f0d55e16ba01ae4e4ca53b823eaa56d6bd2572aa/reana_server/scheduler.py#L141), it would be clear to what actually we do send and we would avoid `user_id` switch with `user`, WDYT?